File: tests/fixtures/go/service.go
package service

import (
	"fmt"

	api "example.com/other/api"
	_ "embed"
)

const MaxRetries = 3

var counter = 0

// Base provides shared plumbing for handlers.
type Base struct {
	Name string
}

// Handler processes jobs from its queue.
type Handler struct {
	Base
	Queue []string
}

// Repo is the storage boundary.
type Repo interface {
	Fetch(id string) ([]byte, error)
}

func (h *Handler) Process(r Repo) error {
	data, err := r.Fetch("job-1")
	if err != nil {
		return fmt.Errorf("fetch failed: %w", err)
	}
	h.store(data)
	return nil
}

func (h *Handler) store(data []byte) error {
	counter++
	return api.Save(MaxRetries, data)
}

func reset() {
	counter = 0
}

func Shutdown(h *Handler) {
	reset()
}

func Run() func() {
	return reset
}

File: tests/fixtures/rust/service.rs
use std::collections::HashMap;
pub use crate::storage::Repo;

pub const MAX_RETRIES: u32 = 3;

pub struct Handler {
    pub queue: Vec<String>,
}

pub trait Storage {
    fn fetch(&self, id: &str) -> Option<String>;
}

impl Handler {
    pub fn process(&self, storage: &dyn Storage) -> Option<String> {
        let data = storage.fetch("job-1");
        self.store(data.clone())
    }

    fn store(&self, data: Option<String>) -> Option<String> {
        data
    }
}

fn reset() {
    let cache: HashMap<String, String> = HashMap::new();
    let _ = cache;
}

pub fn shutdown() {
    reset();
}

pub fn runner() -> fn() {
    reset
}
